### DISCUSSION TITLE
This might not be a bug but a request. Please use any performance-based language like Rust to build reflex CLI.

### DISCUSSION THREAD

**Discussion by itsmeadarsh2008** [reflex-dev/reflex#3031] [Ideas]
I'm posting this request because the hot reload is very delayed in starting up the web app. It installs the same npm modules whenever I start the app. It is very time-consuming. It would be better if we switch to another language maybe **rust** which has become very popular because of its performance in developer tools like CLIs and package-managers. I predict it would be speedy if we switch to Rust replacing the current `reflex run`. It takes too much time to compile.

**Comment by itsmeadarsh2008** [comment]
Convert it to discussion.

**Comment by Alek99** [comment]
I agree compile performance needs to be improved especially for larger apps. We notice it a lot on reflex-web. We do have some way to improve the compile on the python side, specifically involving not re compiling components that haven't changed on every hot reload although its more of an involved switch.

Rust would definitely help. I think we will try and speed it up in Python and see if we hit any blockers with that first.

**Comment by itsmeadarsh2008** [comment]
It's been a long time now, Any Updates?

**Reply by picklelo** [reply]
What's the main issue you're seeing - is it related to compile / hot reload times? We're not planning to integrate Rust or anything, but we are working on performance improvements to make these things faster.

**Reply by itsmeadarsh2008** [reply]
Use libraries like Cython instead of Rust. I'm getting a delay of about 4.5 minutes (hot-reload on the current setup). This may improve performance a lot.

**Reply by masenf** [reply]
4.5 minutes 😱 how big if your app and what is your setup? Even on an underpowered windows machine, running native, i'm seeing 40 seconds for hot-reload from hitting save to the browser showing the change.

And don't get me wrong, 40 seconds sucks, and we do want to improve it, but 4.5 minutes would be intolerable; i don't think that's the norm from what i've seen.

**Reply by itsmeadarsh2008** [reply]
No Nah, It's the first delay of the watcher (it happens with `flet` too, but frameworks like `nicegui` work seamlessly on the first try, maybe Vue is lighter than React). Then, everything works perfectly fine in that 40 seconds but the compilation time sucks so I halted that project because it went from a simple idea to a complex one and required my own Gemini API key for charges which gave me no reason to make it (open-source) one lol. I talked about replacing compilers with faster ones like `esbuild`, `repack` etc. and my setup is a decent Ryzen 5 processor-based HP Laptop with 8Gigs of RAM 😅 (I'm a teenager)
# Please take a look at discussion #3030 about replacing the bundler with faster ones.

![image](https://github.com/reflex-dev/reflex/assets/71959210/07286a4f-089b-4484-a4c9-40e1119faf78)

### `esbuild` does that 0.39 seconds which may not be an issue while developing the web app.
It is my humble request to analyse #3030 (discussion)

**Comment by itsmeadarsh2008** [comment]
Updates?
